File: echo/echo-model/src/main/java/com/netflix/spinnaker/echo/model/ArtifactEvent.java
package com.netflix.spinnaker.echo.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.netflix.spinnaker.echo.api.events.Metadata;
import com.netflix.spinnaker.kork.artifacts.model.Artifact;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ArtifactEvent {
  private Metadata details;
  private List<Artifact> artifacts;
}

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ArtifactEvent.kt
package com.netflix.spinnaker.keel.events

import com.netflix.spinnaker.keel.api.ArtifactStatus
import com.netflix.spinnaker.keel.api.DeliveryArtifact
import com.netflix.spinnaker.kork.artifacts.model.Artifact

data class ArtifactEvent(
  val artifacts: List<Artifact>,
  val details: Map<String, Any>?
)

/**
 * This event optionally takes [statuses] so that we can fetch the most relevant first artifact.
 */
data class ArtifactRegisteredEvent(
  val artifact: DeliveryArtifact,
  val statuses: List<ArtifactStatus> = enumValues<ArtifactStatus>().toList()
)
